docs/source/guides/writer/chapters/recipes.rst

```diff
@@ -76,3 +76,78 @@ That will be parsed by the ``runnables-recipe`` resolver, like in
 
   exec-test /bin/true
   exec-test /bin/false
+
+Using dynamically generated recipes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``exec-runnables-recipe`` resolver allows a user to point to a
+file that will be executed, and that is expected to generate (on its
+``STDOUT``) content compatible with the Runnable recipe format
+mentioned previously.
+
+For security reasons, Avocado won't execute files indiscriminately
+when looking for tests (at the resolution phase). One must set the
+``--resolver-run-executables`` command line option (or the underlying
+``resolver.run_executables`` configuration option) to allow running
+executables at the resolver stage.
+
+.. warning:: It's the user's responsibility to give test references
+             (to be resolved and thus executed) that are well behaved
+             in the sense that they will finish executing quickly,
+             actually execute unintended code (such as running tests),
+             destroying data, etc.
+
+A script such as:
+
+.. literalinclude:: ../../../../../examples/nrunner/resolvers/exec_runnables_recipe.sh
+
+Will output JSON that is compatible with the runnable recipe format.
+That can be used directly via either ``avocado list`` or ``avocado
+run``.  Example::
+
+  $ avocado list --resolver-run-executables examples/nrunner/resolvers/exec_runnables_recipe.sh
+
+  exec-test true-test
+  exec-test false-test
+
+If the executable to be run needs arguments, you can pass it via the
+``--resolver-exec-arguments`` or the underlying
+``resolver.exec_runnable_recipe.arguments`` option.  The following
+script receives an optional parameter that can change the type of the
+tests it generates:
+
+.. literalinclude:: ../../../../../examples/nrunner/resolvers/exec_runnables_recipe_kind.sh
+
+In order to have those tests resolved as ``tap`` tests, one can run::
+
+  $ avocado list --resolver-run-executables --resolver-exec-arguments tap examples/nrunner/resolvers/exec_runnables_recipe_kind.sh
+
+  tap true-test
+  tap false-test
+
+Behavior of ``exec-runnables-recipe`` and ``exec-test`` resolvers
```

Comment:
I am not sure if I can agree with this section. Because, with current behaviour, any executable reference will be executed during resolver phase when the `--resolver-run-executables` is used, which sounds self-explanatory, but IMO it might be confusing for users, because exec-test will be executed twice (once by resolver and second as a test). Therefore, we should let users know, that this is not good practice and that they should split those types into separated suite to avoid this behaviour.

Reply:
I've added the following block to the docs:

```
.. warning:: Even though it's possible to have ``exec-test`` and
             ``exec-runnable-recipes`` in the same Avocado test suite
             (for instance in an ``avocado run`` command execution)
             it's not recommended on most cases because ``exec-tests``
             will end up being run at the test resolution phase
             in addition to the test execution phase.  It's
             recommended to use multiple ``avocado run``
             commands or use the Job API and mulitple
             :class:`avocado.core.suite.TestSuite`, one for
             ``exec-runnable-recipes`` with the
             ``resolver.run_executables`` options enabled, and
             another for ``exec-tests`` with that option in its
             default state (disabled).

```